toolkits/dropbox/arcade_dropbox/tools/browse.py

```diff
@@ -0,0 +1,138 @@
+from typing import Annotated, Optional
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Dropbox
+from arcade.sdk.errors import ToolExecutionError
+
+from arcade_dropbox.enums import Endpoint, ItemCategory
+from arcade_dropbox.exceptions import DropboxApiError
+from arcade_dropbox.utils import (
+    build_dropbox_json,
+    clean_dropbox_entries,
+    parse_dropbox_path,
+    send_dropbox_request,
+)
+
+
+@tool(
+    requires_auth=Dropbox(
+        scopes=["files.metadata.read"],
+    )
+)
+async def list_items_in_folder(
+    context: ToolContext,
+    folder_path: Annotated[
+        str,
+        "The path to the folder to list the contents of. E.g. '/AcmeInc/Reports'. "
+        "Defaults to an empty string (list items in the Dropbox root folder).",
+    ] = "",
+    limit: Annotated[
+        int,
+        "The maximum number of items to return. Defaults to 100. Maximum allowed is 2000.",
+    ] = 100,
+    cursor: Annotated[
+        Optional[str],
+        "The cursor token for the next page of results. "
+        "Defaults to None (returns the first page of results).",
+    ] = None,
+) -> Annotated[
+    dict, "Dictionary containing the list of files and folders in the specified folder path"
+]:
+    """Provides a dictionary containing the list of items in the specified folder path.
+
+    Note 1: when paginating, it is not necessary to provide any other argument besides the cursor.
+    Note 2: when paginating, any given item (file or folder) may be returned in multiple pages.
+    """
+    limit = min(limit, 2000)
+
+    try:
+        result = await send_dropbox_request(
+            None if not context.authorization else context.authorization.token,
```

Comment:
migrate to using [get_auth_token_or_empty](https://github.com/ArcadeAI/arcade-ai/blob/887ded441edbc0c31f8d38be9b1c99adf0887394/arcade/arcade/core/schema.py#L265)